trajectory_executor/com_serial.py

```diff
@@ -0,0 +1,222 @@
+          calculated_checksum += decoded_data
+          read_state = serial_read_states['READ_CHECKSUM']
+        elif read_state == serial_read_states['READ_CHECKSUM']:
+          data_packet['checksum'] = decoded_data
+          if data_packet['checksum'] == 255 - (calculated_checksum % 256):
+            # good packet, relay to tracking node
+            response.roll = data_packet['roll']
+            response.pitch = data_packet['pitch']
+            response.yaw = data_packet['yaw']
+            response.l_hip = data_packet['l_hip']
+            response.l_knee = data_packet['l_hip']
+            response.r_hip = data_packet['r_hip']
+            response.r_knee = data_packet['r_hip']
+            response.l_shoulder = data_packet['l_shoulder']
+            response.l_elbow = data_packet['l_elbow']
+            response.r_shoulder = data_packet['r_shoulder']
+            response.r_elbow = data_packet['r_elbow']
+            response.at_goal = data_packet['at_goal']
+            break
+          else:
+            # bad packet, retry request
+            return self.sensor_data_request_callback(request, response)
+
+    return response
+
+  def send_joint_angles_callback(self, msg: JointAngles):
+    """Send new joint angles to Arduino for execution"""
+    # Split joint angles into two numbers since max value of a byte is 255
+    ja_bytes = []
+    ja_bytes.append(255 if msg.left_hip // 256 > 0 else msg.left_hip)
+    ja_bytes.append(msg.left_hip % 255 if msg.left_hip // 256 > 0 else 0)
+    ja_bytes.append(255 if msg.left_knee // 256 > 0 else msg.left_knee)
+    ja_bytes.append(msg.left_knee % 255 if msg.left_knee // 256 > 0 else 0)
+    ja_bytes.append(255 if msg.right_hip // 256 > 0 else msg.right_hip)
+    ja_bytes.append(msg.right_hip % 255 if msg.right_hip // 256 > 0 else 0)
+    ja_bytes.append(255 if msg.right_knee // 256 > 0 else msg.right_knee)
+    ja_bytes.append(msg.right_knee % 255 if msg.right_knee // 256 > 0 else 0)
+    ja_bytes.append(255 if msg.left_shoulder // 256 > 0 else msg.left_shoulder)
+    ja_bytes.append(msg.left_shoulder % 255 if msg.left_shoulder // 256 > 0 else 0)
+    ja_bytes.append(255 if msg.left_elbow // 256 > 0 else msg.left_elbow)
+    ja_bytes.append(msg.left_elbow % 255 if msg.left_elbow // 256 > 0 else 0)
+    ja_bytes.append(255 if msg.right_shoulder // 256 > 0 else msg.right_shoulder)
+    ja_bytes.append(msg.right_shoulder % 255 if msg.right_shoulder // 256 > 0 else 0)
+    ja_bytes.append(255 if msg.right_elbow // 256 > 0 else msg.right_elbow)
+    ja_bytes.append(msg.right_elbow % 255 if msg.right_elbow // 256 > 0 else 0)
+
+    # calculate checksum
+    checksum = 255 - sum(ja_bytes) % 256
+    ja_bytes.append(checksum)
+
+    # send preamble, joint angles, and checksum
+    for i in range(self.preamble_length):
+      self.arduino_port.write(struct.pack('>B', 255))
+    for ja_byte in ja_bytes:
+      self.arduino_port.write(struct.pack('>B', ja_byte))
```

Comment:
since we only need to be returning joint angles right now, hardcoding the preamble and checksum sandwiching is probably fine. in the future, this should probably be abstracted out cause it's required for all communications.

Reply:
I suppose. Although, I'm in the boat of only abstract if its necessary, and I don't see it as necessary here where there's only two instances of each operation. Sure there's repetition but it's only a few lines so I don't see the need. I we were gonna add more modes of communication that needed to also send preambles/checksums then I would be all for it, but we're very likely only using the current two communication channels.

Reply:
Again let's add this as a code improvement issue with all the necessary details and hash it out later